torch/distributed/_pycute/int_tuple.py

```diff
@@ -39,72 +36,77 @@
 
 from functools import reduce
 from itertools import chain
-from typing import Union
+from typing import Optional, Union
+from typing_extensions import TypeAlias, TypeGuard
 
 from .typing import Integer
 
 
-def is_int(x):
+# Type aliases for better readability
+IntTuple: TypeAlias = Union[int, tuple["IntTuple", ...]]
+
+
+def is_int(x: object) -> TypeGuard[int]:
     return isinstance(x, Integer)
 
 
-def is_tuple(x):
+def is_tuple(x: object) -> TypeGuard[tuple]:
     return isinstance(x, tuple)
 
 
-def flatten(t):
+def flatten(t: IntTuple) -> tuple[int, ...]:
     if is_tuple(t):
         if len(t) == 0:
             return ()
         else:
             return tuple(i for a in t for i in flatten(a))
     else:
-        return (t,)
+        return (t,)  # type: ignore[return-value]  # t is int, converted to tuple[int]
```

Comment:
```suggestion
        return (t,)
```
Wont be necessary with TypeIs